soda_machine.rb

```diff
@@ -7,12 +7,22 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    #@sodas.find {|name| name = soda_brand}
+    #@sodas.select {|name| name = soda_brand}
+    #@sodas.detect{|name| name = soda_brand}
+    #@sodas.index(soda_brand)
+    #these are all the methods I attempted to use, but none passed the tests.
   end
 
   def sell(soda_brand)
+  #@sodas.delete(soda_brand)
+  # @sodas.reject(soda_brand)
+  # I want a method that will return the array without the deleted soda
+  #  @cash += @sodas[:price]
```

Comment:
You do want to add the price of a soda to the cash, but you're treating the `@sodas` array as if it were a Hash.  You need to find the particular `Soda` object, and then get the price from it by calling the `price` method on it.